ee/libc/samples/stdio_tests.c

```diff
@@ -0,0 +1,472 @@
+
+    fp = fopen((char *)arg, "w+");
+    if (fp == NULL)
+    {
+        return "cannot open/create test file";
+    }
+
+    ret = fwrite(buf, 2, 1, fp);
+    if (ret != 1)
+    {
+        fclose(fp);
+        return "failed to write 1 block of 2 bytes";
+    }
+
+    fseek(fp, 0, SEEK_SET);
+    ret = fread(buf2, 2, 1, fp);
+    if (ret != 1 || strncmp(buf2, "he", 2) != 0)
+    {
+        fclose(fp);
+        printf("buf2 = %s ret = %d\n", buf2, ret);
+        return "failed to write/read 2 bytes correctly";
+    }
+
+    ret = fwrite(buf, 1, strlen(buf), fp);
+    if (ret != 11)
+    {
+        fclose(fp);
+        return "failed to write 11 bytes";
+    }
+
+    fseek(fp, 0, SEEK_SET);
+    ret = fread(buf2, 1, sizeof(buf2), fp);
+    if (ret != 13 || strncmp(buf2, "hehello world", 13))
+    {
+        fclose(fp);
+        return "failed to write rest of file";
+    }
+
+    fclose(fp);
+
+    printf("\nSUCCESS: all checks passed\n");
+    return NULL;
+}
+
+static const char *test_stat_file(void *arg)
+{
+    struct stat st;
+
+    if (stat((const char *)arg, &st) != 0)
+    {
+        return "failed to get stat for file";
+    }
+
+    printf("fn %s mode %lo, size %d\n time %s\n",
+           (char *)arg, st.st_mode, (int)st.st_size,
+           ctime(&st.st_mtime));
+
+    if (S_ISDIR(st.st_mode))
+    {
+        return "expected file, not a directory (ignore if using host:)";
+    }
+
+    printf("\nSUCCESS: all checks passed\n");
+    return 0;
+}
+
+static const char *test_stat_dir(void *arg)
+{
+    struct stat st;
+
+    if (stat((const char *)arg, &st) != 0)
+    {
+        return "failed to get stat for directory";
+    }
+
+    printf("fn %s mode %lo, size %d\n time %s\n",
+           (char *)arg, st.st_mode, (int)st.st_size,
+           ctime(&st.st_mtime));
+
+    if (S_ISDIR(st.st_mode) == 0)
+    {
+        return "expected directory, not regular file (ignore if using host:)";
+    }
+
+    printf("\nSUCCESS: all checks passed\n");
+    return 0;
+}
+
+static const char *test_mkdir(void *arg)
+{
+    mode_t mode = 0x0755;
+
+    if (mkdir(arg, mode) != 0)
+    {
+        return "failed to create directory";
+    }
+
+    printf("\nSUCCESS: all checks passed\n");
+    return 0;
+}
+
+static const char *test_rmdir(void *arg)
+{
+    if (rmdir(arg) != 0)
+    {
+        return "failed to delete directory";
+    }
+
+    printf("\nSUCCESS: all checks passed\n");
+    return 0;
+}
+
+static const char *test_opendir_closedir(void *arg)
+{
+    DIR *dd = opendir((char *)arg);
+
+    if (dd == NULL)
+        return "failed to open directory";
+
+    if (closedir(dd) != 0)
+        return "failed to close directory";
+
+    printf("\nSUCCESS: all checks passed\n");
+    return NULL;
+}
+
+static inline int __udelay(unsigned int usecs)
+{
+
+    register unsigned int loops_total = 0;
+    register unsigned int loops_end   = usecs * 148;
+
+    if (usecs > loops_end)
+        return -1;
+
+    __asm__ volatile(".set\tnoreorder\n\t"
```

Comment:
I think that we should use here the delay functions created in `newlib`

Reply:
Totally removed asm code as `sleep()` function is doing the same.